UnifiedPDF: Lookup pill has the wrong font size

hortont424 opened the pull request:
#### 92a81dfc1401cf0f79b97c4123b7251bf4b08dd2
<pre>
UnifiedPDF: Lookup pill has the wrong font size
<a href="https://bugs.webkit.org/show_bug.cgi?id=270835">https://bugs.webkit.org/show_bug.cgi?id=270835</a>
<a href="https://rdar.apple.com/123667947">rdar://123667947</a>

Reviewed by Abrar Rahman Protyasha and Simon Fraser.

Adopt TextIndicator for all UnifiedPDF text indication paths
(Lookup from force-press and from the context menu, in addition to
the existing find-in-page case).

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::showDefinitionForAttributedString):
(WebKit::PDFPlugin::textForImmediateActionHitTestAtPoint):
(WebKit::PDFPlugin::lookupTextAtLocation): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
(WebKit::PDFPluginBase::textIndicatorForCurrentSelection):
(WebKit::PDFPluginBase::textIndicatorForSelection): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::performImmediateActionHitTestAtLocation):
(WebKit::PDFPluginBase::dictionaryPopupInfoForSelection):
Move this PDFPlugin-specific code here from WebPageMac.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::performContextMenuAction):
(WebKit::UnifiedPDFPlugin::textIndicatorForCurrentSelection):
(WebKit::UnifiedPDFPlugin::textIndicatorForSelection):
(WebKit::UnifiedPDFPlugin::performDictionaryLookupAtLocation):
(WebKit::UnifiedPDFPlugin::dictionaryPopupInfoForSelection):
(WebKit::UnifiedPDFPlugin::showDefinitionForSelection):
(WebKit::UnifiedPDFPlugin::textForImmediateActionHitTestAtPoint):
Share more code between find and lookup.

(WebKit::UnifiedPDFPlugin::searchInDictionary): Deleted.
(WebKit::UnifiedPDFPlugin::showDefinitionForAttributedString): Deleted.
(WebKit::UnifiedPDFPlugin::lookupTextAtLocation): Deleted.
* Source/WebKit/WebProcess/Plugins/PluginView.cpp:
(WebKit::PluginView::textIndicatorForCurrentSelection):
(WebKit::PluginView::performImmediateActionHitTestAtLocation const):
(WebKit::PluginView::textIndicatorForSelection): Deleted.
(WebKit::PluginView::lookupTextAtLocation const): Deleted.
* Source/WebKit/WebProcess/Plugins/PluginView.h:
Get rid of LookupTextResult.

* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::updateFindIndicator):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::performImmediateActionHitTestAtLocation):
(WebKit::WebPage::dictionaryPopupInfoForSelectionInPDFPlugin): Deleted.

Canonical link: <a href="https://commits.webkit.org/275994@main">https://commits.webkit.org/275994@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e6b6bccdf80cb08d8304eabf6719d076c06ead63

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43478 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22510 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45889 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46113 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39604 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26305 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19927 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35946 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44052 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19555 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/37461 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16921 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/17110 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1534 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/39656 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38830 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47657 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18508 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15142 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42733 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19931 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41400 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9674 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20110 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19561 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
```diff
@@ -930,6 +931,67 @@
+    CGFloat scaleFactor = contentScaleFactor();
+
+    __block CGFloat maxAscender = 0;
+    __block CGFloat maxDescender = 0;
```
I know you didn't write this originally but do we even need these? I don't see how we actually use them.

aprotyas replied:
Looks like they were used in [162163@main](https://commits.webkit.org/162163@main) but they're not being read anymore.

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
```diff
@@ -930,6 +931,67 @@
+#if PLATFORM(MAC)
+    NSRect rangeRect = rectForSelectionInRootView(selection);
+    NSAttributedString *nsAttributedString = selection.attributedString;
+    RetainPtr<NSMutableAttributedString> scaledNSAttributedString = adoptNS([[NSMutableAttributedString alloc] initWithString:[nsAttributedString string]]);
```
`RetainPtr scaledNSAttributedString =`

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
```diff
@@ -930,6 +931,67 @@
+        NSFont *font = [scaledAttributes objectForKey:NSFontAttributeName];
+        if (font) {
+            maxAscender = std::max(maxAscender, font.ascender * scaleFactor);
+            maxDescender = std::min(maxDescender, font.descender * scaleFactor);
```
The "max = std::min(...)" hurts my brain.

hortont424 committed:
UnifiedPDF: Lookup pill has the wrong font size
https://bugs.webkit.org/show_bug.cgi?id=270835
rdar://123667947

Reviewed by Abrar Rahman Protyasha and Simon Fraser.

Adopt TextIndicator for all UnifiedPDF text indication paths
(Lookup from force-press and from the context menu, in addition to
the existing find-in-page case).

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::showDefinitionForAttributedString):
(WebKit::PDFPlugin::textForImmediateActionHitTestAtPoint):
(WebKit::PDFPlugin::lookupTextAtLocation): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
(WebKit::PDFPluginBase::textIndicatorForCurrentSelection):
(WebKit::PDFPluginBase::textIndicatorForSelection): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::performImmediateActionHitTestAtLocation):
(WebKit::PDFPluginBase::dictionaryPopupInfoForSelection):
Move this PDFPlugin-specific code here from WebPageMac.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::performContextMenuAction):
(WebKit::UnifiedPDFPlugin::textIndicatorForCurrentSelection):
(WebKit::UnifiedPDFPlugin::textIndicatorForSelection):
(WebKit::UnifiedPDFPlugin::performDictionaryLookupAtLocation):
(WebKit::UnifiedPDFPlugin::dictionaryPopupInfoForSelection):
(WebKit::UnifiedPDFPlugin::showDefinitionForSelection):
(WebKit::UnifiedPDFPlugin::textForImmediateActionHitTestAtPoint):
Share more code between find and lookup.

(WebKit::UnifiedPDFPlugin::searchInDictionary): Deleted.
(WebKit::UnifiedPDFPlugin::showDefinitionForAttributedString): Deleted.
(WebKit::UnifiedPDFPlugin::lookupTextAtLocation): Deleted.
* Source/WebKit/WebProcess/Plugins/PluginView.cpp:
(WebKit::PluginView::textIndicatorForCurrentSelection):
(WebKit::PluginView::performImmediateActionHitTestAtLocation const):
(WebKit::PluginView::textIndicatorForSelection): Deleted.
(WebKit::PluginView::lookupTextAtLocation const): Deleted.
* Source/WebKit/WebProcess/Plugins/PluginView.h:
Get rid of LookupTextResult.

* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::updateFindIndicator):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::performImmediateActionHitTestAtLocation):
(WebKit::WebPage::dictionaryPopupInfoForSelectionInPDFPlugin): Deleted.

Canonical link: https://commits.webkit.org/275994@main
